src/main/java/com/lineate/testyourlexicon/repositories/UserRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.lineate.testyourlexicon.repositories;
+
+import com.lineate.testyourlexicon.entities.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+
+  Optional<User> findUserByEmail(String email);
```

Comment:
If we do a search by a column, we'll need to add an index to this column in the database. It is a good practice to add indexes to such columns to avoid possible performance issues.